### PR TITLE
[Method References 2] LPS-177112 Convert method reference to ServiceTracker in classes IndexDefinitionsRegistryImpl, IndexSynchronizationPortalInitializedListener

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:help

**Comment by W2jx** [comment]
ci:test:search

**Comment by zhangyan0701** [comment]
One failure seems related:
Test [pmd-jdk8/0/0](https://testray.liferay.com/home/-/testray/case_results/2218344630) failed with error:
```
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:6806: 1 PMD violation(s) were found.
     [echo] 
     [echo] PMD VIOLATION: /opt/dev/projects/github/liferay-portal/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/index/instant/InstantIndexesTest.java:151 > Do not call Assert#fail improperly. (PMD Rule: AssertFailJUnitRule)
```
Full console:
https://testray.liferay.com/reports/production/logs/2023-03/test-1-32/test-portal-acceptance-pullrequest(master)/8432/pmd-jdk8/0/0/jenkins-console.txt.gz?authuser=0

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:search

**Comment by W2jx** [comment]
no conflicts(3/6 9:05)

**Comment by W2jx** [comment]
no conflicts(3/7 9:02)

**Comment by W2jx** [comment]
no conflicts(3/8 15:25)

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/IndexSynchronizer.java [hunk 1]
```diff
@@ -24,8 +24,6 @@ public interface IndexSynchronizer {
 	public void synchronizeIndexDefinition(
 		IndexDefinitionData indexDefinitionData);
 
-	public void synchronizeIndexes();
```
Comment (tinatian) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/IndexSynchronizer.java:27]: @W2jx I do not understand why a merge needs to remove an API method. Please separate your changes into two commits.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/IndexSynchronizer.java [hunk 2]
```diff
@@ -14,16 +14,151 @@
...
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
 
 /**
  * @author André de Oliveira
  */
-public interface IndexSynchronizer {
+@Component(service = {})
```
Comment (tinatian) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/IndexSynchronizer.java:47]: @W2jx When you merge, it is better you do not change logic, it is really hard to find.

Here, before merge, it is "@Component(immediate = true, service = IndexSynchronizer.class)", and now it is "@Component(service = {})"

**Comment by tinatian** [comment]
@W2jx Please see my comment and resend.

Thanks.
Tina

**Pull request closed without merging**